core/src/main/java/org/apache/iceberg/util/TableScanUtil.java

```diff
@@ -78,4 +83,57 @@ public static CloseableIterable<CombinedScanTask> planTasks(CloseableIterable<Fi
             splitFiles),
         BaseCombinedScanTask::new);
   }
+
+  public static <T extends SplittableScanTask<T>> CloseableIterable<InputSplit<T>> planInputSplits(
```

Comment:
The existing logic we had for splitting and combining expressed in terms of `SplittableScanTask`.